tfhe/src/core_crypto/algorithms/test/mod.rs

```diff
@@ -225,4 +363,26 @@ macro_rules! create_parametrized_test{
     };
 }
 
+#[cfg(feature = "__coverage")]
+macro_rules! create_parametrized_test{
+    ($name:ident { $($param:ident),* }) => {
+        paste! {
+            $(
+            #[test]
+            fn [<test_ $name _ $param:lower>]() {
+                $name($param)
+            }
+            )*
+        }
+    };
+     ($name:ident)=> {
+        create_parametrized_test!($name
+        {
+            TEST_PARAMS_4_BITS_NATIVE_U64
```

Comment:
this is lowering coverage vs. the other create parameterized tests

Reply:
Adding the other parameters set will probably dramatically increase coverage duration.

Reply:
I'm running the early break currently to see how long it last.

Reply:
potentially the avx512 does not need to re run both parameter sets

Reply:
but it starts to make a very convoluted test set up

Reply:
Weirdly, the early break gives us little gains, around 30 min.
I see some tests that can be improved, especially the ones related to key generation.

Does using the second parameters set will improve coverage ? I.e is the code taking other branches in this case ?

Reply:
yes

Reply:
I'll add the other parameter set and run the test to check how long it takes.

Reply:
don't forget the keygen

but I think we need to identify the slowest tests and limit the runs of those

Reply:
The cost of adding the second parameters set is negligible. It also improve coverage a bit. :tada: